proposals/12513-pvc-artifact-storage/README.md

```diff
@@ -0,0 +1,1867 @@
+|--------------------------|------------------------------------------------------------------------------|---------------------------------------------------------------|
+| **URI** (stored in MLMD) | `kfp-artifacts://<namespace>/<pipeline>/<run-id>/<node-id>/<artifact-name>`  | `kfp-artifacts://team-a/my-pipeline/abc123/step1/output`      |
+| **API Endpoint**         | `/apis/v2beta1/runs/{run_id}/nodes/{node_id}/artifacts/{artifact_name}:read` | `/apis/v2beta1/runs/abc123/nodes/step1/artifacts/output:read` |
+| **Filesystem Path**      | `<mount_path>/<namespace>/<pipeline>/<run-id>/<node-id>/<artifact-name>`     | `/artifacts/team-a/my-pipeline/abc123/step1/output`           |
+
+**Why the API endpoint doesn't include namespace/pipeline:**
+
+The API endpoint only requires `run_id`, `node_id`, and `artifact_name`. The server resolves the full filesystem path by looking up the run's metadata from the database (which contains the namespace and pipeline name). This matches the existing artifact server behavior for S3/GCS.
+
+**How the Launcher uses the URI:**
+
+1. Launcher receives artifact URI: `kfp-artifacts://team-a/my-pipeline/abc123/step1/output`
+2. Extracts `run_id=abc123`, `node_id=step1`, `artifact_name=output`
+3. In namespace-local mode, extracts `namespace=team-a` to route to correct artifact server
+4. Makes API call: `GET http://ml-pipeline-artifact-server.team-a.svc:8080/apis/v2beta1/runs/abc123/nodes/step1/artifacts/output:read`
+
+This approach provides several benefits:
+
+1. **Abstraction**: Clients don't need to know the underlying storage backend
+2. **Security**: All artifact access goes through KFP's authorization layer  
+3. **Scalability**: Artifact serving can be scaled independently
+4. **Flexibility**: Storage backend can be changed without client modifications
+
+### Artifact Server Architecture
+
+#### Backend Responsibilities
+
+KFP will support two distinct deployment modes for artifact serving. The default mode is configured globally via the `ObjectStoreConfig.ArtifactServer.DeploymentMode` field, but individual namespaces can override this setting via their `kfp-launcher` ConfigMap. **In both modes, only the artifact server pods mount the PVC - pipeline workflow pods access artifacts exclusively through the artifact server API.**
+
+##### Mode 1: Central Artifact Server (Default)
+
+A single artifact server in the main KFP namespace serves all namespaces, configured via `ObjectStoreConfig.ArtifactServer.DeploymentMode: "central"`.
+
+**Central Mode Characteristics:**
+
+- **Single PVC** with directory structure: `/artifacts/<namespace>/<pipeline>/<run-id>/<node-id>/<artifact-name>`
+- **Authorization**: Uses `SubjectAccessReview` to verify namespace access
+- **Best for**: Simple deployments, single-user setups, small teams
+- **Advantages**: Simple setup, single storage location, easy backup
+- **Limitations**: All namespaces share same PVC and storage quota
```

Comment:
```suggestion
```

For the reasons mentioned above and below this mode has been made fully obsolete by the default multi-tenant seaweedfs. Furthermore it is a security nightmare and adds unnecessary complexity. There is zero benefit implementing this.